client/ayon_blender/plugins/load/load_image_compositor.py

```diff
@@ -0,0 +1,150 @@
+        path = self.filepath_from_context(context)
+        img_comp_node = container["node"]
+
+        old_image: Optional[bpy.types.Image] = img_comp_node.image
+
+        new_image = bpy.data.images.load(path, check_existing=True)
+        img_comp_node.image = new_image
+
+        self.set_source_and_colorspace(context, img_comp_node)
+        self.remove_image_if_unused(old_image)
+
+        # Update representation id
+        lib.imprint(img_comp_node, {
+            "representation": context["representation"]["id"]
+        })
+
+    def set_source_and_colorspace(
+        self,
+        context: dict,
+        image_comp_node: bpy.types.CompositorNodeImage
+    ):
+        """
+        Set the image source (e.g. SEQUENCE or FILE), set the duration for
+        a sequence and set colorspace if representation has colorspace data.
+        """
+
+        image = image_comp_node.image
+        representation: dict = context["representation"]
+
+        # Set image source
+        source = "FILE"  # Single image file
+        if representation["context"].get("udim"):
+            source = "UDIM"
+        elif representation["context"].get("frame"):
+            source = "SEQUENCE"
+        else:
+            ext = os.path.splitext(image.filepath)[-1]
+            if ext in VIDEO_EXTENSIONS:
+                source = "MOVIE"
+
+        image.source = source
+
+        # Set duration on the compositor node if sequence is used
+        if source in {"SEQUENCE", "MOVIE"}:
+            version_attrib: dict = context["version"]["attrib"]
+            frame_start = version_attrib.get("frameStart", 0)
+            frame_end = version_attrib.get("frameEnd", 0)
+            handle_start = version_attrib.get("handleStart", 0)
+            handle_end = version_attrib.get("handleEnd", 0)
+            frame_start_handle = frame_start - handle_start
+            frame_end_handle = frame_end + handle_end
+            duration: int = frame_end_handle - frame_start_handle + 1
+            image_comp_node.frame_duration = duration
+            if source == "SEQUENCE":
+                image_comp_node.frame_start = frame_start_handle
+                image_comp_node.frame_offset = frame_start_handle - 1
+            else:
+                image_comp_node.frame_start = frame_start_handle
+                image_comp_node.frame_offset = 0
+
+        # Set colorspace if representation has colorspace data
+        if representation.get("colorspaceData"):
+            colorspace: str = representation["colorspaceData"]["colorspace"]
+            if colorspace:
+                image.colorspace_settings.name = colorspace
+
+    def remove_image_if_unused(self, image: bpy.types.Image):
+        if image and not image.users:
+            self.log.debug("Removing unused image: %s", image.name)
+            bpy.data.images.remove(image)
+
+    def switch(self, container, context):
+        # Support switch in scene inventory
+        self.update(container, context)
```

Comment:
honestly didn't see the switch function in the blender loaders.  Should we create separate PR to add this back?

Reply:
Moved that to https://github.com/ynput/ayon-blender/pull/43